test/reflection.jl

```diff
@@ -262,6 +262,10 @@ tlayout = TLayout(5,7,11)
 @test fieldtype((NamedTuple{(:a,:b),T} where T<:Tuple{Vararg{Integer}}), 2) === Integer
 @test_throws BoundsError fieldtype(NamedTuple{(:a,:b)}, 3)
 
+@test fieldtypes(NamedTuple{(:a,:b)}) == (Any, Any)
+@test fieldtypes((NamedTuple{T,Tuple{Int,String}} where T)) === (Int, String)
+@test fieldtypes(TLayout) === (Int8, Int16, Int32)
```

Comment:
Can we test that these are inferred correctly? Or are they inferred as `Tuple{Datatype,Datatype,...}`?

Reply:
Would adding an `@inferred` be a satisfactory test? Because that passes.